builder/azure/arm/config.go

```diff
@@ -303,6 +303,8 @@ func setCloudEnvironment(c *Config) error {
 	switch name {
 	case "CHINA", "CHINACLOUD", "AZURECHINACLOUD":
 		c.cloudEnvironment = &azure.ChinaCloud
+	case "GERMAN", "GERMANCLOUD", "AZUREGERMANCLOUD":
```

Comment:
I'm -1 on having multiple names for clouds. It's better to be precise/consistent than forgiving.